### DISCUSSION TITLE
How to capture keyboard event CMD + Q in MacOs or Alt + F4 in Windows

### DISCUSSION THREAD

**Discussion by totaland** [tauri-apps/tauri#8341] [Q&A]
tauri::Builder::default()
    .on_window_event(|event| match event.event() {
      WindowEvent::Destroyed => {
        Command::new("killall")
        .arg("node");
      }
      _ => {}
    })

I tried to run command to terminate all node processes when user quit with CMD + Q but this is failed.

What are the ways to do this? Much appreciate your help.

**Comment by FabianLars** [comment]
Alt + F4 should trigger CloseRequested & Destroyed and ExitRequested & Exit just fine.
CMD + Q however is a special apple event that tauri does not yet handle, at least for the Window events, i'd recommend to try out Exit and ExitRequest (these are app, not window events): https://docs.rs/tauri/latest/tauri/struct.App.html#method.run - I _think_ they also don't work but worth a try.

**Reply by totaland** [reply]
Thanks man for the information. I will give it a try. I feels the same it might not work.

**Reply by Sytten** [reply]
They dont work I can confirm, is this something that is planned to be improved?
